### ISSUE TITLE
Need an easy way to install examples

### ISSUE THREAD

**Issue by SGeeversAtVortech** [Deltares/rtc-tools#1033]
In GitLab by @baayen on May 17, 2018, 14:14

The 2.0 installer included an 'examples' folder.

For 2.2, a 'pip install rtc-tools-examples' would be nice.

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 22, 2018, 24:24

I think a pip install would not be very helpful. First of all, it will be quite difficult to package something like that with the correct version numbering, and with all the package data. From playing around a bit, I don't think this is a case that setuptools/distutils cover well.

More importantly though, using pip install would still put the examples in a folder that the user probably cannot find. Put differently, the user that can find the folder with the Python package can probably also locate the examples on GitLab.

Instead, I would like to propose using a (somewhat hackish, but more userfriendly approach) by making a small program that does this. E.g. when you install rtc-tools, you can then run `rtctools download_examples`. It would then just download the source zip corresponding to the RTC-Tools version from GitLab, and extract just the examples (and move them to a folder named "rtc-tools-examples" in the current working directory).

```python
import shutil
import os
from urllib.error import HTTPError
from urllib.request import urlretrieve
from zipfile import ZipFile

import rtctools

version = rtctools.__version__
rtc_full_name = 'rtc-tools-{}'.format(version)
try:
    urlretrieve(
        'https://gitlab.com/deltares/rtc-tools/-/archive/'
        '{}/{}.zip'.format(version, rtc_full_name),
        'rtctools.zip')
except HTTPError:
    raise Exception("Could not found examples for RTC-Tools version {}.".format(version))

with ZipFile('rtctools.zip', 'r') as z:
    prefix = '{}/examples/'.format(rtc_full_name)
    members = [x for x in z.namelist() if x.startswith(prefix)]
    z.extractall(members=members)
    shutil.move(prefix, 'rtc-tools-examples')
    shutil.rmtree(rtc_full_name)

os.remove('rtctools.zip')
```

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 22, 2018, 23:27

mentioned in commit 93c1f8811c678a8474b1684176b09343a4f02b16

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 26, 2018, 11:51

mentioned in commit 1b8810dfa511bd5d755a62a253f4b60cf2f78d4b

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 26, 2018, 12:04

mentioned in commit 576e88187444f066add68bfdb029c03219c6f120

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 26, 2018, 12:11

mentioned in commit 752fb4243f8ea67db9c7a9c5841c8cac4c1e139f

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 26, 2018, 12:18

closed via merge request !122

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 26, 2018, 12:18

closed via commit 752fb4243f8ea67db9c7a9c5841c8cac4c1e139f
